tracing-log/src/log_tracer.rs

```diff
@@ -0,0 +1,232 @@
+use crate::{format_trace, AsTrace};
+use log;
+use tracing_core::dispatcher;
+
+/// A simple "logger" that converts all log records into `tracing` `Event`s.
+///
+/// Can be initialized with:
+///
+/// * [`init`] if you want to convert all logs and do the filtering in a subscriber
+/// * [`init_with_filter`] if you know in advance a log level you want to filter
+///
+/// [`init`]: ../fn.init.html
+/// [`init_with_filter`]: ../fn.init_with_filter.html
+#[derive(Debug)]
+pub struct LogTracer {
+    ignore_crates: Box<[String]>,
+}
+
+#[derive(Debug)]
+pub struct Builder {
+    ignore_crates: Vec<String>,
+    filter: log::LevelFilter,
+}
+
+// ===== impl LogTracer =====
+
+impl LogTracer {
+    /// Returns a builder that allows customizing a `LogTracer` and setting it
+    /// the default logger.
+    ///
+    /// For example:
+    /// ```rust
+    /// # use std::error::Error;
+    /// use tracing_log::LogTracer;
+    /// use log;
+    ///
+    /// # fn main() -> Result<(), Box<Error>> {
+    /// LogTracer::builder()
+    ///     .ignore_crate("foo") // suppose the `foo` crate is using `tracing`'s log feature
+    ///     .with_max_level(log::LevelFilter::Info)
+    ///     .init()?;
+    ///
+    /// // will be available for Subscribers as a tracing Event
+    /// log::info!("an example info log");
+    /// # Ok(())
+    /// # }
+    /// ```
+    pub fn builder() -> Builder {
+        Builder::default()
+    }
+
+    /// Creates a new `LogTracer` that can then be used as a logger for the `log` crate.
+    ///
+    /// It is generally simpler to use the [`init`] or [`init_with_filter`] methods
+    /// which will create the `LogTracer` and set it as the global logger.
+    ///
+    /// Logger setup without the initialization methods can be done with:
+    ///
+    /// ```rust
+    /// # use std::error::Error;
+    /// use tracing_log::LogTracer;
+    /// use log;
+    ///
+    /// # fn main() -> Result<(), Box<Error>> {
+    /// let logger = LogTracer::new();
+    /// log::set_boxed_logger(Box::new(logger))?;
+    /// log::set_max_level(log::LevelFilter::Trace);
+    ///
+    /// // will be available for Subscribers as a tracing Event
+    /// log::trace!("an example trace log");
+    /// # Ok(())
+    /// # }
+    /// ```
+    ///
+    /// [`init`]: #method.init
+    /// [`init_with_filter`]: .#method.init_with_filter
+    pub fn new() -> Self {
+        Self {
+            ignore_crates: Vec::new().into_boxed_slice(),
+        }
+    }
+
+    /// Sets up `LogTracer` as global logger for the `log` crate,
+    /// with the given level as max level filter.
+    ///
+    /// Setting a global logger can only be done once.
+    ///
+    /// The [`builder`] function can be used to customize the `LogTracer` before
+    /// initializing it.
+    ///
+    /// [`builder`]: #method.builder
+    pub fn init_with_filter(level: log::LevelFilter) -> Result<(), log::SetLoggerError> {
```

Comment:
Do you think it makes sense to keep this method on `LogTracer` now it can be done with the builder? (and maybe for the `init` method too).

Reply:
I left it because I thought it was better not to break any code currently using this method.

Reply:
Ok :+1: